### PR TITLE
Feature/image upload

### PR DESCRIPTION
**Pull request by natashaval**
presence -> upload image -> log check in
1. Face accepted
2. Face Not Match

profile -> fetch from API -> save to Room

### CODE REVIEW

File: Clockio/app/src/main/java/com/natasha/clockio/presence/repository/ImageRepositoryImpl.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import com.natasha.clockio.presence.util.CallbackAggregator
+import java.util.*
+
+class ImageRepositoryImpl: ImageRepository {
+
+  private val TAG: String = "ImageRepository"
+
+  override fun uploadImage(userId: String, imagePath: String) {
+    MediaManager.get().upload(imagePath)
+        .unsigned("zmfwv7yt")
```
Comment (cesarioputera) [inline Clockio/app/src/main/java/com/natasha/clockio/presence/repository/ImageRepositoryImpl.kt]: put these in  Constants

File: Clockio/app/src/main/java/com/natasha/clockio/presence/ui/fragment/ImageFragment.kt [hunk 2]
```diff
@@ -1,65 +1,202 @@
...
+    observeProgress()
+    observeEvents()
+    observeCheckInResult()
+  }
 
-    private lateinit var viewModel: ImageViewModel
-    private lateinit var imagePath: String
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    employeeId = sharedPref.getString("id", null)
```
Comment (cesarioputera) [inline Clockio/app/src/main/java/com/natasha/clockio/presence/ui/fragment/ImageFragment.kt]: put this String in Constants, so other files can access these also

File: Clockio/app/src/main/java/com/natasha/clockio/presence/repository/ImageRepositoryImpl.kt (hunk 1 above)
Comment (natashaval) [reply Clockio/app/src/main/java/com/natasha/clockio/presence/repository/ImageRepositoryImpl.kt]: done

File: Clockio/app/src/main/java/com/natasha/clockio/presence/ui/fragment/ImageFragment.kt (hunk 2 above)
Comment (natashaval) [reply Clockio/app/src/main/java/com/natasha/clockio/presence/ui/fragment/ImageFragment.kt]: done

File: Clockio/app/src/main/java/com/natasha/clockio/presence/ui/fragment/ImageFragment.kt [hunk 3]
```diff
@@ -64,20 +69,18 @@ class ImageFragment : Fragment() {
...
+    observeLocation()
     observeProgress()
     observeEvents()
     observeCheckInResult()
   }
 
   override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
     super.onViewCreated(view, savedInstanceState)
-    employeeId = sharedPref.getString("id", null)
+    employeeId = sharedPref.getString(getString(R.string.employee_id_key), null)
```
Comment (cesarioputera) [inline Clockio/app/src/main/java/com/natasha/clockio/presence/ui/fragment/ImageFragment.kt]: make a Constant class for this

File: Clockio/app/src/main/java/com/natasha/clockio/presence/repository/ImageRepositoryImpl.kt [hunk 4]
```diff
@@ -1,20 +1,23 @@
...
 
-class ImageRepositoryImpl: ImageRepository {
+class ImageRepositoryImpl @Inject constructor(private val context: Context): ImageRepository {
 
   private val TAG: String = "ImageRepository"
 
   override fun uploadImage(userId: String, imagePath: String) {
     MediaManager.get().upload(imagePath)
-        .unsigned("zmfwv7yt")
+        .unsigned(context.resources.getString(R.string.cloudinary_unsigned))
```
Comment (cesarioputera) [inline Clockio/app/src/main/java/com/natasha/clockio/presence/repository/ImageRepositoryImpl.kt]: make a Constant class for this

File: Clockio/app/src/main/java/com/natasha/clockio/presence/ui/fragment/ImageFragment.kt (hunk 3 above)
Comment (natashaval) [reply Clockio/app/src/main/java/com/natasha/clockio/presence/ui/fragment/ImageFragment.kt]: done changed to object const

File: Clockio/app/src/main/java/com/natasha/clockio/presence/repository/ImageRepositoryImpl.kt (hunk 4 above)
Comment (natashaval) [reply Clockio/app/src/main/java/com/natasha/clockio/presence/repository/ImageRepositoryImpl.kt]: done changed to object const

**Pull request merged**